model/distro/distro.go

```diff
@@ -537,12 +534,28 @@ func (d *Distro) RemoveExtraneousProviderSettings(region string) {
 			"region":        region,
 			"settings_list": d.ProviderSettingsList,
 		}))
-		return
+		return errors.Wrapf(err, "error getting provider settings by region")
 	}
 	d.ProviderSettingsList = []*birch.Document{doc}
+	d.ProviderSettings = nil
+	return nil
 }
 
 func (d *Distro) GetProviderSettingByRegion(region string) (*birch.Document, error) {
+	if (region == "" || region == evergreen.DefaultEC2Region) && len(d.ProviderSettingsList) == 0 {
```

Comment:
There are 2 region == "" checks in a row. Is that intentional?

Reply:
Yeah, I could comment better: the first one is for if `len(d.ProviderSettingsList) == 0` and the other one is not.